core/services/ocr2/plugins/median/services.go

```diff
@@ -152,3 +180,49 @@ func NewMedianServices(ctx context.Context,
 	}
 	return
 }
+
+type medianContract struct {
+	chainReader types.ChainReader
+	contract    types.BoundContract
+}
+
+type latestTransmissionDetailsResponse struct {
+	configDigest    ocr2types.ConfigDigest
+	epoch           uint32
+	round           uint8
+	latestAnswer    *big.Int
+	latestTimestamp time.Time
+}
+
+type latestRoundRequested struct {
+	configDigest ocr2types.ConfigDigest
+	epoch        uint32
+	round        uint8
+}
+
+func (m *medianContract) LatestTransmissionDetails(ctx context.Context) (configDigest ocr2types.ConfigDigest, epoch uint32, round uint8, latestAnswer *big.Int, latestTimestamp time.Time, err error) {
+	var resp latestTransmissionDetailsResponse
+
+	err = m.chainReader.GetLatestValue(ctx, m.contract, "LatestTransmissionDetails", map[string]string{}, &resp)
```

Comment:
Is passing `nil` allowed?
```suggestion
	err = m.chainReader.GetLatestValue(ctx, m.contract, "LatestTransmissionDetails", nil, &resp)
```
Even if the micro-optimization of skipping the allocation is not worth it, supporting nil could prevent runtime bugs, and follows the Go idiom of treating nil and empty slices/maps the same.

Reply:
I figured an empty map would be safer to transport across process boundaries than nil, since in json, sql, and most other languages nil and empty map are seen as different types. (And even in go, checking to see if a map is empty could cause a runtime bug which would be avoided if it's an empty map instead of nil) But I don't see any specific problem with having it nil, so I guess we can go with that.  (Out of curiousity, what runtime bugs would nil prevent? Off the top of my head, I can only think of runtime bugs it could cause.)

Reply:
I mean that supporting nil gracefully would eliminate the possibility of accidentally causing a runtime nil panic.